odc/geo/cog/_az.py

```diff
@@ -1,37 +1,73 @@
 import base64
+import sys
+import threading
+import time
+import uuid
 from typing import Any
+from dask.distributed import get_client, Lock
+from dask.base import tokenize
 
+from azure.core.exceptions import AzureError, HttpResponseError
+from azure.storage.blob import BlobBlock, BlobServiceClient, ContentSettings
 from ._multipart import MultiPartUploadBase
 
+import logging
+
+logger = logging.getLogger("odc.geo.cog._az")
+# Set default level (can be overridden by application config)
+logger.setLevel(logging.INFO)
+# Avoid adding handlers multiple times if this module is reloaded
+if not logger.hasHandlers():
+    # Configure console handler
+    handler = logging.StreamHandler(sys.stdout)
+    formatter = logging.Formatter(
+        "%(asctime)s [%(levelname)s] %(name)s (%(funcName)s): %(message)s"
+    )
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    # Prevent logs from propagating to the root logger if handlers are added here
+    logger.propagate = False
+
+# Example: Increase log level for debugging
+# logger.setLevel(logging.DEBUG)
+
 
```

Comment:
no logging setup like that on import please, do it inside your test app if needed for debugging